File: tests/test_runtime.py
import json
from pathlib import Path
import sys

ROOT = Path(__file__).resolve().parents[1]
sys.path.insert(0, str(ROOT / "python"))

from misp_validation import RuleEngine

engine = RuleEngine.from_file(ROOT / "spec" / "attributes.json")
vectors = json.loads((ROOT / "tests" / "vectors.json").read_text(encoding="utf-8"))

for vector in vectors:
    result = engine.validate(vector["type"], vector["input"])
    assert result.valid == vector["valid"], (vector, result)
    assert result.value == vector["normalized"], (vector, result)

print(f"OK: {len(vectors)} validation vectors")

File: tests/test_runtime.php
<?php

declare(strict_types=1);

require dirname(__DIR__) . '/php/MispValidation/RuleEngine.php';

use MispValidation\RuleEngine;

$root = dirname(__DIR__);
$engine = RuleEngine::fromFile($root . '/spec/attributes.json');
$vectors = json_decode(file_get_contents($root . '/tests/vectors.json'), true, flags: JSON_THROW_ON_ERROR);

foreach ($vectors as $index => $vector) {
    $result = $engine->validate($vector['type'], $vector['input']);
    assert($result->valid === $vector['valid'], "vector {$index}: validity differs");
    assert($result->value === $vector['normalized'], "vector {$index}: normalized value differs");
}

echo 'OK: ' . count($vectors) . " validation vectors\n";
